packages/grafana-ui/src/components/Sparkline/Sparkline.tsx

```diff
@@ -103,25 +103,30 @@ export class Sparkline extends PureComponent<SparklineProps, State> {
 
   getYRange(field: Field): Range.MinMax {
     let { min, max } = this.state.alignedDataFrame.fields[1].state?.range!;
-    const noValue = +this.state.alignedDataFrame.fields[1].config?.noValue!;
 
+    // enure that the min/max from the field config are respected
+    min = Math.max(min!, field.config.min ?? -Infinity);
+    max = Math.min(max!, field.config.max ?? Infinity);
+
+    // if noValue is set, ensure that it is included in the range as well
+    const noValue = +this.state.alignedDataFrame.fields[1].config?.noValue!;
     if (!Number.isNaN(noValue)) {
-      min = Math.min(min!, +noValue);
-      max = Math.max(max!, +noValue);
+      min = Math.min(min, +noValue);
+      max = Math.max(max, +noValue);
```

Comment:
doesn't matter what order we do the first two steps in here, it just made more sense to me to apply the field config min and max first.